ConfigSpace/c_util.pyx

```diff
@@ -266,83 +204,64 @@ cpdef np.ndarray change_hp_value(
     """
     cdef Hyperparameter current
     cdef str current_name
-    cdef list disabled
-    cdef set visited
-    cdef dict activated_values
     cdef int active
+    cdef int update
     cdef ConditionComponent condition
     cdef int current_idx
     cdef DTYPE_t current_value
+    cdef DTYPE_t target_value
     cdef DTYPE_t default_value
-    cdef list children
-    cdef list children_
-    cdef Hyperparameter ch
-    cdef str child
-    cdef set to_disable
     cdef DTYPE_t NaN = np.NaN
     cdef dict children_of = configuration_space._children_of
 
-    configuration_array[index] = hp_value
-
-    # Hyperparameters which are going to be set to inactive
-    disabled = []
+    # We maintain to_visit as a minimum heap of indices of hyperparameters that may need to be updated.
+    # We assume that the hyperparameters are sorted topologically by their index.
+    to_visit = [index]
+    
+    # Since one hyperparameter may be reachable in several ways, we need to make sure we don't process it twice.
+    scheduled = np.zeros(len(configuration_space), dtype=bool)
```

Comment:
`scheduled` is a replacement of `visited`. While `visited` is a set of HP names, `scheduled` is a boolean vector representation of a set of HPs, where each HP is identified by its index. I estimate that this change is probably a premature potential optimization.